packages/fxa-settings/src/pages/InlineRecoverySetup/container.tsx

```diff
@@ -0,0 +1,116 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { useMutation } from '@apollo/client';
+import { RouteComponentProps, useLocation, useNavigate } from '@reach/router';
+import LoadingSpinner from 'fxa-react/components/LoadingSpinner';
+import { useCallback, useEffect, useState } from 'react';
+import InlineRecoverySetup from './index';
+import { AuthUiErrors } from '../../lib/auth-errors/auth-errors';
+// import { useFinishOAuthFlowHandler } from '../../lib/oauth/hooks';
+import { getCode } from '../../lib/totp';
+import { MozServices } from '../../lib/types';
+import {
+  OAuthIntegration,
+  useAccount,
+  // useAuthClient,
+  useSession,
+} from '../../models';
+import { TotpToken } from '../InlineTotpSetup';
+import { VERIFY_TOTP_MUTATION } from './gql';
+
+export const InlineRecoverySetupContainer = ({
+  isSignedIn,
+  integration,
+  serviceName,
+}: {
+  isSignedIn: boolean;
+  integration: OAuthIntegration;
+  serviceName: MozServices;
+} & RouteComponentProps) => {
+  const account = useAccount();
+  const session = useSession();
+  const navigate = useNavigate();
+
+  // const authClient = useAuthClient();
+  // const { finishOAuthFlowHandler, oAuthDataError } = useFinishOAuthFlowHandler(
+  //   authClient,
+  //   integration
+  // );
+
+  const location = useLocation() as ReturnType<typeof useLocation> & {
+    state: {
+      totp: TotpToken;
+    };
+  };
+  const totp = location.state?.totp;
+
+  const [recoveryCodes, setRecoveryCodes] = useState<string[]>();
+  const [verifyTotp] = useMutation<{ verifyTotp: { success: boolean } }>(
+    VERIFY_TOTP_MUTATION
+  );
+
+  const verifyTotpHandler = useCallback(async () => {
+    const code = await getCode(totp.secret);
+    const result = await verifyTotp({
```

Comment:
Drive by question for general understanding (not changes in this PR) - this mutation uses the account resolver and is in the account model. Should it instead be in the session resolver since it uses the `session/verify` endpoint?

Reply:
I started moving the verifyTotp mutation to the SessionResolver but then thought it was odd to have createTotp and deleteTotp in AccountResolver while having verifyTotp in SessionResolver so I opted to keep all three in AccountResolver for now.

Reply:
Oh one other thing. My initial comment alluded to the fact that verfyTotp in the account model still uses auth-client, which might actually be better until we fix the IP forwarding issue... but long term, using GQL seems like the way we are going.